internal/controller/kyma/controller.go

```diff
@@ -45,7 +45,7 @@ import (
 	"github.com/kyma-project/lifecycle-manager/pkg/status"
 	"github.com/kyma-project/lifecycle-manager/pkg/templatelookup"
 	"github.com/kyma-project/lifecycle-manager/pkg/util"
-	"github.com/kyma-project/lifecycle-manager/pkg/watcher"
+	"github.com/kyma-project/lifecycle-manager/pkg/watcher/certificate"
```

Comment:
Although it's a valid usage to depend on sub package `watcher/certificate`, since it's in controller layer, I want to have a clear dependency here, which means the controller only depends on business layer (Service layer), and Watcher should be the public interface to controller level, the certificate remain encapsulated within watcher packge.

Reply:
This is an interesting case. Let's first align on the other dependency-related comments and then get back to this.

Reply:
Moved the error to `watcher`: fa24b93 See also comment below